pkg/stanza/operator/parser/keyvalue/keyvalue.go

```diff
@@ -122,14 +122,45 @@ func (kv *Parser) parser(input string, delimiter string) (map[string]any, error)
 	return parsed, err
 }
 
-// split on whitespace and preserve quoted text
-func splitStringByWhitespace(input string) []string {
-	quoted := false
-	raw := strings.FieldsFunc(input, func(r rune) bool {
-		if r == '"' || r == '\'' {
-			quoted = !quoted
+// splitPairs will split the input on the pairDelimiter and return the resulting slice.
+// `strings.Split` is not used because it does not respect quotes and will split if the delimiter appears in a quoted value
+func splitPairs(input, pairDelimiter string) ([]string, error) {
+	var result []string
+	currentPair := ""
+	delimiterLength := len(pairDelimiter)
+	quoteChar := "" // "" means we are not in quotes
+
+	i := 0
+	for i < len(input) {
+		if quoteChar == "" && i+delimiterLength <= len(input) && input[i:i+delimiterLength] == pairDelimiter {
```

Comment:
This seems a bit more complicated than necessary. Could you iterate on this some more. My advice would be:
1. As noted in other comments, you only need to increment `i` manually when you find a delimiter. Otherwise, just let the loop handle it.
2. Try to handle "quick conclusions" first, so you can just if `if simple case { continue }`. This flattens the deeply nested structure because afterwards you know any tokens did not match those conditions.

Reply:
Updated the loop to be an actual for loop now and flattened most of the nested `if` blocks. Let me know if I should be more aggressive in simplifying the loop or if this new version is good!